### ISSUE TITLE
Support Confluence Cloud REST API v2

### ISSUE THREAD

**Issue by jdknight** [sphinx-contrib/confluencebuilder#858]
At the time of writing, it has been over a year since Atlassian releases an initial REST API v2 for Confluence. This extension should be taking steps to support the new REST API version; ideally, supporting both, defaulting to v2 calls for Cloud, but allowing users to configure which to use (in case of issues). There has been some talk [1](https://community.developer.atlassian.com/t/rfc-19-deprecation-of-confluence-cloud-rest-api-v1-endpoints/71752/22) [2](https://community.developer.atlassian.com/t/i-am-sorry-is-there-a-new-deadline-for-api-v1-to-be-migrated-to-v2/74194/12) about how deprecated v1 endpoints may no longer be supported in the upcoming months.

**Comment by jdknight** [comment]
Reference: https://developer.atlassian.com/cloud/confluence/changelog/#CHANGE-864

**Comment by jdknight** [comment]
See also #913.

**Comment by jdknight** [comment]
The initial implementation needed to support REST API v2 has now been merged into the development branch. Limited testing appears to work well, but more testing should be done before we produce a v2.5 release.

Confluence Cloud's changelog last indicates that June 2024 is the cut-off date before needing to use API v2 calls. It would be nice to complete a v2.5 release a month before this date to give some time for any issues that may arise from a v2.5 release -- allowing time to create a v2.5.1 release while blocked users can just force using an older API version or using a legacy extension version.

**Comment by jdknight** [comment]
v2.5 is now available on PyPI -- marking as closed.
